### PR TITLE
RDF generation changes to better match wikidata RDF dumps

### PR DESCRIPTION
**Pull request by bennofs**
This PR includes a few small changes that make the Wikidata RDF export more similar to what's in the dumps. 

Some of these changes are backwards-incompatible, such as the change to how SomeValue claims are exported or the IRI changes. Not sure how we want to deal with that.

The statement id / reference hash changes are not strictly necessary, but also simple and it's nice if the subject IRIs match the full dumps.

### CODE REVIEW

**Comment by wetneb** [comment]
IMHO it is totally fine to be "backwards incompatible" in this way: the RDF that is generated by Wikibase itself is the standard, so any difference between WDTK's RDF and Wikibase's RDF is a bug, that is therefore worth fixing. (But I am not using wdtk-rdf so @Tpt will be better placed to review this).

File: wdtk-rdf/src/main/java/org/wikidata/wdtk/rdf/Vocabulary.java [hunk 1]
```diff
@@ -493,8 +490,9 @@
 	 */
 	public static String getStatementUri(Statement statement) {
 		int i = statement.getStatementId().indexOf('$') + 1;
+		final String id = i <= 0 ? statement.getSubject().getId() : statement.getStatementId().substring(0, i-1);
 		return PREFIX_WIKIDATA_STATEMENT
-				+ statement.getSubject().getId() + "-"
+				+ id + "-"
 				+ statement.getStatementId().substring(i);
```
[the remark this answers is not in the document]
Comment (bennofs) [reply wdtk-rdf/src/main/java/org/wikidata/wdtk/rdf/Vocabulary.java]: A statement id does not always contain a `$`, for example the tests frequently pass just "" as statement id. In that case, we should still include the item id in the serialized statement for clarity. This requires checking if `$` is part of the statement, so we would have a condition in any case. We could do `statement.getStatementId().contains("$")` and only do the replacement if that's true, but would that be much clearer? Happy to change it to that version if you want

[the remark this answers is not in the document]
Comment (Tpt) [reply wdtk-rdf/src/main/java/org/wikidata/wdtk/rdf/Vocabulary.java]: If there is no statement id (or just ""), adding the entity id adds some clarity indeed but does not solve an underlying problem: if there are multiple statements with id "" on the same item, they are going to end up with the same statement IRI, leading to a bad RDF representation.

A simpler way to solve this problem would be to use `org.wikidata.wdtk.wikibaseapi.RandomGuidGenerator` to generate a statement id if the provided statement id is null or `""` and do the simple replacement if not. What do you think about it?

**Comment by Tpt** [comment]
@bennofs Is it fine for you if I take over this pull request to fix the tests before merging it?

**Comment by bennofs** [comment]
@Tpt yes of course

**Comment by Tpt** [comment]
Improved in #449

**Pull request closed without merging**